src/main/java/com/itrex/konoplyanik/boardgamerent/controller/handler/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,61 @@
+package com.itrex.konoplyanik.boardgamerent.controller.handler;
+
+import com.itrex.konoplyanik.boardgamerent.exception.ServiceException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@Slf4j
+@ControllerAdvice
+public class ControllerExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseEntity<?> handleMethodArgumentTypeMismatchException(
+            WebRequest request,
+            MethodArgumentTypeMismatchException ex
+    ) {
+        log.error("handleMethodArgumentTypeMismatchException {}\n", request.getDescription(false), ex);
+        return handleExceptionInternal(ex,"Method argument type mismatch", new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
```

Comment:
Very interesting way, I somehow didn’t realize that you can invoke the existing method of parent class